lib/engine/step/tracker.rb

```diff
@@ -60,6 +60,7 @@ def lay_tile_action(action, entity: nil, spender: nil)
           raise GameError, "#{action.hex.id} cannot be layed as this hex was already layed on this turn"
         end
 
+        tile_lay_cost_override!(tile_lay, action, tile, old_tile)
```

Comment:
Depending on what you are trying to do, there are already ways to set the tile lay costs. Can you please provide detail on what the game needs to do, so I can steer you in the right direction.